### ISSUE TITLE
Profiled GPU and memory are all zeros

### ISSUE THREAD

**Issue by Olliebrown** [apitrace/apitrace#359]
With glretrace, if I turn on gpu and memory profiling I do not get any error messages but nothing but zeros are output for gpu and memory values.  The gpu start and duration are all zeros as are vsize start/dura and rss start/dura.  CPU timings profile fine as do pixel counts (after latest commit).

System details:
- Mid 2014 Macbook Pro running OSX 10.9.5
- GeForce GT 750M
- Traced program was using OpenGL 4.1 Core profile

Some example output:
```
c-12293:ReflectanceAcquisition berriers$ /Users/berriers/Programming/apitrace/apitrace/build/glretrace --pgpu --pcpu --ppd --pmem java.trace 
# call no gpu_start gpu_dura cpu_start cpu_dura vsize_start vsize_dura rss_start rss_dura pixels program name
call 7 0 0 0 0 0 0 0 0 -1 0 glViewport
call 8 0 0 0 0 0 0 0 0 -1 0 glScissor
call 175 0 0 0 0 0 0 0 0 0 0 glClear
frame_end
... skipping to important frames ...
frame_end
call 425183 0 0 15833487394 1666 0 0 0 0 -1 1 glBindFramebuffer
call 425185 0 0 15833497097 51061 0 0 0 0 -1 1 glClearBufferfv
call 425189 0 0 15833558829 14045 0 0 0 0 -1 1 glClearBufferfv
call 425191 0 0 15833578757 4294 0 0 0 0 -1 1 glGetUniformLocation
call 425195 0 0 15833591715 4949 0 0 0 0 -1 1 glUseProgram
call 425199 0 0 15833604062 1218 0 0 0 0 -1 1 glBindTexture
call 425517 0 0 15833993609 1680 0 0 0 0 -1 1 glBindBufferBase
call 425657 0 0 15834197668 1346 0 0 0 0 -1 1 glUniform1i
call 425659 0 0 15834202887 7032 0 0 0 0 -1 1 glGetUniformBlockIndex
call 425663 0 0 15834217921 6379 0 0 0 0 -1 1 glGetUniformBlockIndex
call 425667 0 0 15834231015 3162 0 0 0 0 -1 1 glGetUniformBlockIndex
call 425675 0 0 15834248045 1048 0 0 0 0 -1 1 glUseProgram
call 429883 0 0 15838045076 1744 0 0 0 0 -1 1 glBindVertexArray
call 429885 0 0 15838052435 45296970 0 0 0 0 725059 1 glDrawArrays
call 429887 0 0 15883359252 10170051 0 0 0 0 -1 1 glFlush
frame_end
call 429890 0 0 16066326800 1692 0 0 0 0 -1 1 glBindFramebuffer
call 429892 0 0 16066336304 53830 0 0 0 0 -1 1 glClearBufferfv
call 429896 0 0 16066401244 14435 0 0 0 0 -1 1 glClearBufferfv
call 429898 0 0 16066421897 4621 0 0 0 0 -1 1 glGetUniformLocation
call 429902 0 0 16066435296 5252 0 0 0 0 -1 1 glUseProgram
call 429906 0 0 16066448092 1205 0 0 0 0 -1 1 glBindTexture
call 430224 0 0 16066840428 1814 0 0 0 0 -1 1 glBindBufferBase
call 430364 0 0 16067051165 1794 0 0 0 0 -1 1 glUniform1i
call 430366 0 0 16067057084 7363 0 0 0 0 -1 1 glGetUniformBlockIndex
call 430370 0 0 16067072718 6339 0 0 0 0 -1 1 glGetUniformBlockIndex
call 430374 0 0 16067085542 3171 0 0 0 0 -1 1 glGetUniformBlockIndex
call 430382 0 0 16067102710 1050 0 0 0 0 -1 1 glUseProgram
call 430844 0 0 16068404185 1163 0 0 0 0 -1 1 glUniform1i
call 430846 0 0 16068408662 1389 0 0 0 0 -1 1 glGetUniformLocation
call 430850 0 0 16068416586 1209 0 0 0 0 -1 1 glUseProgram
call 431780 0 0 16069450784 1072 0 0 0 0 -1 1 glUniformMatrix4fv
call 434590 0 0 16071911219 1671 0 0 0 0 -1 1 glBindVertexArray
call 434592 0 0 16071919916 45384955 0 0 0 0 725059 1 glDrawArrays
call 434594 0 0 16117317876 9707261 0 0 0 0 -1 1 glFlush
... truncated ...
```

**Comment by jrfonseca** [comment]
Seems to be a known problem on Macs per http://lists.apple.com/archives/mac-opengl/2014/Nov/msg00001.html .

Elapsed times still appear to work.  I pushed a workaround that just enables elapsed times.

**Comment by jrfonseca** [comment]
>  nothing but zeros are output for gpu and memory values

Oh, concerning memory, it hasn't been implemented for anything bug Linux.  But this measures CPU memory.

We could implement on other platforms, or add better diagnostic, but I'm not sure how much use this is for GL app developers.

Something like GPU memory (using vendor specific extensions) would probably more interesting, but that's a whole different topic.

**Comment by Olliebrown** [comment]
Ooo, bummer.  Well thank you for all your help today!  Hopefully these fixes will help others too.

**Comment by Olliebrown** [comment]
Just wanted to follow up on this.  Your workaround did not quite solve the problem for me.  It was still getting 0 for all the GPU duration times.  In the end, I found a way to get the numbers I was missing.

In TraceProfiler.cpp, the method addCall(...), right near the beginning (currently around line 115), it tests to see if gpuStart is not zero and subtracts it if it is not.  However, if it is zero (and GPU timing is enabled) it always sets it to zero AND sets gpuDuration to zero.  I simply removed this last step (commented out line 119 where it sets gpuDuration to 0) and then I was getting GPU duration numbers in the output where I expected.

I'm sure a more nuanced fix is needed for the long run but just thought I would follow up with my experience.
